File: 1184. Distance Between Bus Stops/1184.go
package  leetcode

func distanceBetweenBusStops(distance []int, start int, destination int) int {
	if start > destination{
		destination, start = start, destination
	}

	var tmp1, tmp2 int
	for i:=start; i<destination; i++{
		tmp1 += distance[i]
	}

	for i:=0; i<start; i++{
		tmp2 += distance[i]
	}
	for i:=destination; i<len(distance); i++{
		tmp2 += distance[i]
	}

	if tmp1>tmp2{
		return tmp2
	}
	return tmp1
}

File: 1184. Distance Between Bus Stops/1184.py
class Solution:
    def distanceBetweenBusStops(self, distance, start: int, destination: int) -> int:
        start, destination = min(start, destination), max(start, destination)
        return min(sum(distance[start:destination]), (sum(distance[:start])+sum(distance[destination:])))

a = Solution()
print(a.distanceBetweenBusStops(distance = [1,2,3,4], start = 0, destination = 3))